state/indexer/block/kv/kv.go

```diff
@@ -648,28 +653,35 @@ func (idx *BlockerIndexer) indexEvents(batch dbm.Batch, events []abci.Event, hei
 				if err := batch.Set(key, heightBz); err != nil {
 					return err
 				}
+				keyArray = AppendToKeyArray(keyArray, key)
 			}
 		}
 	}
+	if err := batch.Set(getEventsForHeightKey(height), keyArray); err != nil {
```

Comment:
Instead of introducing a new key, you can actually reuse the already existing height key. In the block indexer you can [see](https://github.com/cometbft/cometbft/blob/1e0cf136d395db94728883355a136e4e34e693db/state/indexer/block/kv/kv.go#L77) that the height key has the heigth as well as the value. So you can easily replace it.  You can use the same key which will overwrite the previous value stored.